server/bundler/extensions-loader.js

```diff
@@ -0,0 +1,74 @@
+const fs = require( 'fs' );
+const path = require( 'path' );
+const camelCase = require( 'lodash' ).camelCase;
+const compact = require( 'lodash' ).compact;
+
+function getReducerPath( extensionDir ) {
+	return path.join( 'client', 'extensions', extensionDir, 'state', 'reducer.js' );
+}
+
+function generateReducerRequireString( extensionDir ) {
+	return `'${ camelCase( extensionDir ) }': require( 'extensions/${ extensionDir }/state/reducer' )\n`;
+}
+
+function generateExtensionsModuleString( reducerRequires ) {
+	return `module.exports = {
+		reducers: function() {
+			return {
+				${ reducerRequires }
```

Comment:
I know that the VM takes care of joining the array's lines when interpolating into the string template, but I'd play it safe and explicitly join:
```js
`
			return {
				${ reducerRequires.join( separator ) }
`
```

If you choose `separator := '\n'`, you can ditch the newline in `generateReducerRequireString`.